src/libraries/Microsoft.PowerFx.Core/IR/IRTranslator.cs

```diff
@@ -663,7 +663,8 @@ public override IntermediateNode Visit(FirstNameNode node, IRTranslatorContext c
                         }
 
                     case BindKind.OptionSet:
-                    case BindKind.PowerFxResolvedObject:
+                    case BindKind.PowerFxResolvedObject:
+                    case BindKind.Data:
                         {
```

Comment:
What is the scenario for this?

Reply:
This was necessary to support a new `TableValue` derived class (`EntityTableValue`). This new class simulates a database table and has `AssociatedDataSource` within its type.